### PR TITLE
Handle start and stop actions from widgets

### LINKED ISSUE #5086: Handle start and stop actions
Call the API to start and stop actions, the API client instance moved to `ExtensionKit` contains all we need to start and stop time entries.

Check `StartTimerIntentHandler` and `StopTimerIntentHandler` in the Siri extension for reference.

Depends on #5085 and #5500

### PR DESCRIPTION
**Pull request by dshr**
## What's this?
Teaches our widgets how to call the API and start/stop TEs

### Relationships
Closes #5086

## Why do we want this?
For the greater good

## How is it done?
By adding a dedicated networking handler which will take care of all our networking needs.

## :squid: Permissions
🎛 people only, and only after this is not a draft.

### CODE REVIEW

File: Toggl.iOS.TimerWidgetExtension/TodayViewController.cs [hunk 1]
```diff
@@ -1,13 +1,17 @@
 ﻿using System;
-using NotificationCenter;
 using Foundation;
+using NotificationCenter;
```
Comment (juanlaube) [inline Toggl.iOS.TimerWidgetExtension/TodayViewController.cs]: 👀

File: Toggl.iOS.TimerWidgetExtension/TodayViewController.cs [hunk 2]
```diff
@@ -16,6 +20,13 @@ public override void ViewDidAppear(bool animated)
         {
             base.ViewDidAppear(animated);
             SharedStorage.Instance.SetWidgetUpdatedDate(DateTimeOffset.Now);
+
+            StopButton.AddTarget(stopTimeEntry, UIControlEvent.TouchUpInside);
```
Comment (juanlaube) [inline Toggl.iOS.TimerWidgetExtension/TodayViewController.cs]: This should be in `viewDidLoad`, right? Also, no start action? 🙂

File: Toggl.iOS.TimerWidgetExtension/TodayViewController.cs (hunk 1 above)
Comment (dshr) [reply Toggl.iOS.TimerWidgetExtension/TodayViewController.cs]: That's needed for `WidgetPerformUpdate`and `INCWidgetProviding`

File: Toggl.iOS.TimerWidgetExtension/TodayViewController.cs (hunk 2 above)
Comment (dshr) [reply Toggl.iOS.TimerWidgetExtension/TodayViewController.cs]: D'oh! Completely missed the `ViewDidAppear`. As for the start action — isn't that dependent on #5087?

**Pull request merged**